init.js

```diff
@@ -48,6 +46,80 @@ function reset() {
     oldTotals.parentNode.replaceChild(newTotals, oldTotals)
 }
 
+function loadItems(settings) {
+    // clear current targets
+    build_targets = []
+    var targetList = document.getElementById("targets")
+    targetList.innerHTML = targetList.lastChild.outerHTML
+
+    if ("items" in settings && settings.items != "") {
+        var targets = settings.items.split(",")
+        for (var i=0; i < targets.length; i++) {
+            var targetString = targets[i]
+            var parts = targetString.split(":")
+            var name = parts[0]
+            var target = addTarget(name)
+            var type = parts[1]
+            if (type == "f") {
+                target.setFactories(parts[2])
+            } else if (type == "r") {
+                target.setRate(parts[2])
+            } else {
+                throw new Error("unknown target type")
+            }
+        }
+    } else {
+        addTarget()
+    }
+}
+
+function loadModules(settings) {
```

Comment:
On further review, I was wrong earlier: This function does not properly reset things to their defaults from a given settings object. It will need some rewriting, but I'd be okay applying that fix after accepting the rest of this PR.

This does mean that the change, as written, won't quite handle modules correctly.